### PR TITLE
Add relatedImages to ClusterServiceVersion

### PR DESCRIPTION
**Pull request by bmangoen**
We want to add `relatedImages` to the Sail Operator `ClusterServiceVersion` ( cf. #216 )

### CODE REVIEW

File: Makefile.core.mk [hunk 1]
```diff
@@ -497,6 +497,8 @@ bundle: gen helm operator-sdk ## Generate bundle manifests and metadata, then va
 				git checkout "$$csvPath" || echo "failed to revert timestamp change. assuming we're in the middle of a merge"; \
 			fi \
 		fi
+
+	@hack/patch-csv.sh bundle/manifests/$(OPERATOR_NAME).clusterserviceversion.yaml
```
Comment (dgn) [inline Makefile.core.mk:501]: If you do this earlier (between `helm template` and `operator-sdk generate bundle`) you can probably avoid the reformatting

Comment (bmangoen) [reply Makefile.core.mk:501]: You're right! 
Even just between `helm template ... | operator-sdk generate bundle` and `operator-sdk generate kustomize manifests` (cf. [here](https://github.com/istio-ecosystem/sail-operator/blob/main/Makefile.core.mk#L486-L492)) should work

**Review by FilipB — CHANGES_REQUESTED** [review]
I tried to run the script in our fork and it adds olm.relatedImage fields to already existing [images](https://github.com/openshift-service-mesh/sail-operator/blob/main/bundle/manifests/servicemeshoperator3.clusterserviceversion.yaml#L514) fields which is not correct. I'm not sure what is the best solution but we need to make sure this change will not break downstream builds

**Comment by dgn** [comment]
> I tried to run the script in our fork and it adds olm.relatedImage fields to already existing [images](https://github.com/openshift-service-mesh/sail-operator/blob/main/bundle/manifests/servicemeshoperator3.clusterserviceversion.yaml#L514) fields which is not correct. I'm not sure what is the best solution but we need to make sure this change will not break downstream builds

maybe we can add an env var to control whether the related images are added? then we can disable it in the fork

**Comment by bmangoen** [comment]
@FilipB @dgn `make -e GENERATE_RELATED_IMAGES=false gen` should not generate the annotations `olm.relatedImage.<version>.<component_name>` and the `relatedImages` part

**Comment by dgn** [comment]
> @FilipB @dgn `make -e GENERATE_RELATED_IMAGES=false gen` should not generate the annotations `olm.relatedImage.<version>.<component_name>` and the `relatedImages` part

The annotations should stay - but it looks to me like they will?

**Review by dgn — APPROVED** [review]

**Review by FilipB — APPROVED** [review]

**Comment by FilipB** [comment]
Added https://github.com/openshift-service-mesh/sail-operator/pull/112 to disable this in our fork.

**Pull request merged**